### PR TITLE
GH-55 Improve extension integration with PHPStan

### LINKED ISSUE #55: Improve the extension
Hi, I'd like to discourage you from using the CompilerExtension concept. The fact that PHPStan uses Nette\DI is not a public API and can break. Also, it means that this PHPStan extension is not usable with phpstan-shim (PHAR) where Nette is prefixed.

Also, I will remove the global variables in the next version, I didn't realize that someone would access and read them intentionally - they're not part of PHPStan's official API :)

### PR DESCRIPTION
**Pull request by mglaman**
Fixes #55

### CODE REVIEW

**Comment by attrib** [comment]
I'm currently testing this branch with drupal upgrade_status.

The first issue I run into is `Fatal error: Cannot declare class Drupal, because the name is already in use in /Users/karl.fritsche/workspace/drupal_contrib/drupal/core/lib/Drupal.php on line 80`

This is due to `PHPStan\Drupal\Bootstrap::addCoreNamespaces()` does a `require` to Drupal.php
When the code is run inside the Drupal Context this is already provided and fails with the above error message.

Changing it to a require_once works.

The first run passed in my case, but continue to review+test with the `upgrade_status` module.

**Comment by attrib** [comment]
Last comment + the branch would fix

https://www.drupal.org/project/upgrade_status/issues/3057853
https://www.drupal.org/project/upgrade_status/issues/3050653

From my testing this looks good, with the minor change posted above.

**Comment by mglaman** [comment]
@attrib We should not need to require `Drupal.php` per https://github.com/drupal/core/blob/8.6.x/composer.json#L216.

🤔 or does this change the operation of the loaded autoloading files.

File: src/DependencyInjection/DrupalExtension.php [hunk 1]
```diff
@@ -95,6 +75,12 @@ public function loadConfiguration(): void
                 }
             }
         }
+        return;
```
[the remark this answers is not in the document]
Comment (attrib) [reply src/DependencyInjection/DrupalExtension.php]: Got it now, code was moved to Bootstrap.php, code after return should be removed then.

**Comment by attrib** [comment]
> @attrib We should not need to require `Drupal.php` per https://github.com/drupal/core/blob/8.6.x/composer.json#L216.
> 
> 🤔 or does this change the operation of the loaded autoloading files.

Was this maybe added for drupal-check then?

**Comment by mglaman** [comment]
Refactoring this in #91

**Pull request closed without merging**